conans/client/tools/settings.py

```diff
@@ -0,0 +1,35 @@
+from conans.errors import ConanInvalidConfiguration
+from conans.client.build.cppstd_flags import cppstd_flag, cppstd_from_settings
+from conans.client.tools.oss import OSInfo
+
+
+def cppstd_minimum_required(conanfile, cppstd, gnu_extensions=False):
+    """ Validate if current cppstd fits the minimal version required.
+
+        In case the current cppstd doesn't fit the minimal version required
+        by cppstd, a ConanInvalidConfiguration exception will be raised.
+
+    :param conanfile: ConanFile instance with cppstd to be compared
+    :param cppstd: Minimal cppstd version required
+    :param gnu_extensions: GNU extension is required (e.g gnu17). This option ONLY works on Linux.
+    """
+    def extract_cpp_version(cppstd):
+        return int(str(cppstd).replace("gnu", ""))
```

Comment:
What if we compare as string replacing all the non numerical chars? Could it be a better approach to manage any setting value?

Reply:
do you want to avoid any convert exception?

Reply:
done

Reply:
done what?

Reply:
I've removed the int conversion.

Reply:
My bad. github showed old commits.